### PR TITLE
[FVM] capture events of meta transactions

### PR DESCRIPTION
**Pull request by ramtinms**
This fixes the issue of fee deduction not emitting events.

### CODE REVIEW

File: fvm/transactionFeeDeductor.go [hunk 1]
```diff
@@ -37,14 +36,24 @@ func (d *TransactionFeeDeductor) Process(
 func (d *TransactionFeeDeductor) deductFees(
 	vm *VirtualMachine,
 	ctx *Context,
-	tx *flow.TransactionBody,
+	proc *TransactionProcedure,
```
Comment (m4ksio) [inline fvm/transactionFeeDeductor.go:39]: Hmm, not sure if this is right change. Isn't `TransactionProcedure` supposed to be used only once

File: fvm/transactionFeeDeductor.go [hunk 2]
```diff
@@ -37,14 +36,24 @@ func (d *TransactionFeeDeductor) Process(
...
 		*ctx,
-		deductTransactionFeeTransaction(tx.Payer, ctx.Chain.ServiceAddress()),
+		feeTx,
 		sth,
 		programs,
 	)
+	if txErr == nil {
+		proc.Events = append(proc.Events, feeTx.Events...)
+		proc.ServiceEvents = append(proc.ServiceEvents, feeTx.ServiceEvents...)
+		proc.Logs = append(proc.Logs, feeTx.Logs...)
```
Comment (m4ksio) [inline fvm/transactionFeeDeductor.go:54]: Shouldn't we collect this regardless?

File: fvm/transactionFeeDeductor.go [hunk 3]
```diff
@@ -37,14 +36,24 @@ func (d *TransactionFeeDeductor) Process(
...
+		proc.GasUsed = proc.GasUsed + feeTx.GasUsed
```
Comment (m4ksio) [inline fvm/transactionFeeDeductor.go:55]: and gas used as well?

**Review by m4ksio — APPROVED** [review]

File: fvm/transactionFeeDeductor.go (hunk 1 above)
Comment (ramtinms) [reply fvm/transactionFeeDeductor.go:39]: yeah, this is a temp measure to be able to return the events back to the original caller through this proc, Fees are going to be refactored soon by Janez, so basically this should suffice to fix the current issue.

File: fvm/transactionFeeDeductor.go (hunk 2 above)
Comment (ramtinms) [reply fvm/transactionFeeDeductor.go:54]: we can't right now, we don't have a way to discard some events, I'm working on an event collector handler with the functionality to purge some of the events similar to the state behavior. but right now this is consistent with how we capture events inside the transaction procedure.

**Review by Kay-Zee — APPROVED** [review]

**Pull request merged**